add example requests for endpoints that do not use bodies

miguelgrinberg opened the pull request:
This PR adds missing examples of ES endpoints that do not use bodies. These are used to generate the `x-codeSamples` extension for OpenAPI.

Some examples were reconstructucted from description fields used in response examples (when endpoints have them). For endpoints where there is no response example an example was obtained from an old 8.x alternatives report JSON file.

In a previous PR I added `method_request` fields to some responses. These are now unnecessary so they were removed.

I have also removed the `x-codeSamples` overlay files. In a follow up change I will be adding the client languages for all examples, using the request converter to generate them.

Preview: https://www.elastic.co/docs/api/doc/elasticsearch/version/475682a4-c607-4ca5-a63c-26c1f9b636da/

With all these changes, only the following endpoints are missing examples:

```
HEAD /
GET /_cat
POST /_cluster/voting_config_exclusions
DELETE /_cluster/voting_config_exclusions
HEAD /_component_template/{name}
POST /_connector
PUT /_connector/{connector_id}/_filtering/_activate
PUT /_connector/{connector_id}/_filtering/_validation
PUT /_connector/{connector_id}/_native
GET /_data_stream/{name}/_options
PUT /_data_stream/{name}/_options
DELETE /_data_stream/{name}/_options
GET /_data_stream/{name}/_settings
PUT /_data_stream/{name}/_settings
GET /_fleet/_fleet_msearch
POST /_fleet/_fleet_msearch
HEAD /_index_template/{name}
POST /_inference/{inference_id}
POST /_inference/{task_type}/{inference_id}
GET /_ingest/geoip/database
GET /_ingest/geoip/database/{id}
PUT /_ingest/geoip/database/{id}
DELETE /_ingest/geoip/database/{id}
POST /_ml/anomaly_detectors/{job_id}/_data
PUT /_ml/trained_models/{model_id}
GET /_nodes/{node_id}/_repositories_metering
DELETE /_nodes/{node_id}/_repositories_metering/{max_archive_version}
GET /_query/queries
GET /_query/queries/{id}
POST /_security/delegate_pki
GET /_security/enroll/node
POST /_slm/stop
GET /{index}/_fleet/_fleet_msearch
POST /{index}/_fleet/_fleet_msearch
GET /{index}/_fleet/_fleet_search
POST /{index}/_fleet/_fleet_search
GET /{index}/_fleet/global_checkpoints
```

miguelgrinberg commented on docs/overlays/elasticsearch-shared-overlays.yaml:
```diff
@@ -1077,17 +1077,6 @@ actions:
   #   update:
   #     aggregations:
   #       x-model: true
-# Examples
```
Thanks, Lisa! I forgot to look for this. :)

lcawl commented:
> In a previous PR I added method_request fields to same responses. These are now unnecessary so they were removed.

Is it very simple to understand now when that info is required? e.g. is it only required in at least one request or response body file (and only in the one that we want to appear in the console/curl section)? And if there are multiple request or response body examples that have this field, which one is picked for use in x-codeSamples? IMO we should try to make this as clear and simple as possible (and if that means we leave the `method_request` in all of them to avoid confusion, that's fine by me).

lcawl commented on specification/ml/put_data_frame_analytics/examples/request/MlPutDataFrameAnalyticsExample1.yaml:
```diff
@@ -0,0 +1,41 @@
+      "training_percent": 90
+      }
+    },
+    "analyzed_fields": { <5>
```
```suggestion
    "analyzed_fields": {
```

lcawl commented on specification/nodes/reload_secure_settings/examples/request/NodesReloadSecureSettingsExample1.yaml:
```diff
@@ -0,0 +1,6 @@
+description: An example body for a `POST _nodes/reload_secure_settings` request.
+value: |-
+  {
+    "secure_settings_password": "keystore-password" <1>
```
```suggestion
    "secure_settings_password": "keystore-password"
```

lcawl commented on specification/searchable_snapshots/mount/examples/request/SearchableSnapshotsMountExample1.yaml:
```diff
@@ -0,0 +1,11 @@
+description: An example body for a `POST /_snapshot/my_repository/my_snapshot/_mount?wait_for_completion=true` request.
+value: |-
+  {
+    "index": "my_docs", <1>
```
```suggestion
    "index": "my_docs",
```

lcawl commented on specification/searchable_snapshots/mount/examples/request/SearchableSnapshotsMountExample1.yaml:
```diff
@@ -0,0 +1,11 @@
+value: |-
+  {
+    "index": "my_docs", <1>
+    "renamed_index": "docs", <2>
```
```suggestion
    "renamed_index": "docs",
```

lcawl commented on specification/searchable_snapshots/mount/examples/request/SearchableSnapshotsMountExample1.yaml:
```diff
@@ -0,0 +1,11 @@
+  {
+    "index": "my_docs", <1>
+    "renamed_index": "docs", <2>
+    "index_settings": { <3>
```
```suggestion
    "index_settings": {
```

lcawl commented on specification/searchable_snapshots/mount/examples/request/SearchableSnapshotsMountExample1.yaml:
```diff
@@ -0,0 +1,11 @@
+    "index_settings": { <3>
+      "index.number_of_replicas": 0
+    },
+    "ignore_index_settings": [ "index.refresh_interval" ] <4>
```
```suggestion
    "ignore_index_settings": [ "index.refresh_interval" ]
```

lcawl commented on specification/security/create_api_key/examples/request/SecurityCreateApiKeyExample1.yaml:
```diff
@@ -0,0 +1,20 @@
+      "my-restricted-role-descriptor": {
+        "indices": [
+          {
+            "names": ["website-product-search"], <1>
```
```suggestion
            "names": ["website-product-search"],
```

lcawl commented on specification/security/create_api_key/examples/request/SecurityCreateApiKeyExample1.yaml:
```diff
@@ -0,0 +1,20 @@
+          }
+        ],
+        "restriction":  {
+          "workflows": ["search_application_query"] <2>
```
```suggestion
          "workflows": ["search_application_query"]
```

lcawl commented on specification/security/create_cross_cluster_api_key/examples/request/SecurityCreateCrossClusterApiKeyExample1.yaml:
```diff
@@ -0,0 +1,27 @@
+value: |-
+  {
+    "name": "my-cross-cluster-api-key",
+    "expiration": "1d",   <1>
```
```suggestion
    "expiration": "1d",
```

lcawl commented on specification/security/create_cross_cluster_api_key/examples/request/SecurityCreateCrossClusterApiKeyExample1.yaml:
```diff
@@ -0,0 +1,27 @@
+    "name": "my-cross-cluster-api-key",
+    "expiration": "1d",   <1>
+    "access": {
+      "search": [  <2>
```
```suggestion
      "search": [
```

lcawl commented on specification/security/create_cross_cluster_api_key/examples/request/SecurityCreateCrossClusterApiKeyExample1.yaml:
```diff
@@ -0,0 +1,27 @@
+          "names": ["logs*"]
+        }
+      ],
+      "replication": [  <3>
```
```suggestion
      "replication": [
```

lcawl commented on specification/security/put_privileges/examples/request/SecurityPutPrivilegesExample1.yaml:
```diff
@@ -0,0 +1,15 @@
+  {
+    "myapp": {
+      "read": {
+        "actions": [ <1>
```
```suggestion
        "actions": [
```

lcawl commented on specification/security/put_privileges/examples/request/SecurityPutPrivilegesExample1.yaml:
```diff
@@ -0,0 +1,15 @@
+    "myapp": {
+      "read": {
+        "actions": [ <1>
+          "data:read/*" , <2>
```
```suggestion
          "data:read/*",
```

lcawl commented on specification/security/put_privileges/examples/request/SecurityPutPrivilegesExample1.yaml:
```diff
@@ -0,0 +1,15 @@
+        "actions": [ <1>
+          "data:read/*" , <2>
+          "action:login" ],
+          "metadata": { <3>
```
```suggestion
          "metadata": {
```

lcawl commented on specification/security/put_role_mapping/examples/request/SecurityPutRoleMappingExample1.yaml:
```diff
@@ -0,0 +1,10 @@
+  {
+    "roles" : [ "monitoring" , "user" ],
+    "rules" : { "field" : {
+      "groups" : "cn=admins,dc=example,dc=com" <1>
```
```suggestion
      "groups" : "cn=admins,dc=example,dc=com"
```

lcawl commented on specification/security/suggest_user_profiles/examples/request/SecuritySuggestUserProfilesExample1.yaml:
```diff
@@ -0,0 +1,15 @@
+description: An example body for a `POST /_security/profile/_suggest` request.
+value: |-
+  {
+    "name": "jack",  <1>
```
```suggestion
    "name": "jack",
```

lcawl commented on specification/security/suggest_user_profiles/examples/request/SecuritySuggestUserProfilesExample1.yaml:
```diff
@@ -0,0 +1,15 @@
+  {
+    "name": "jack",  <1>
+    "hint": {
+      "uids": [  <2>
```
```suggestion
      "uids": [
```

lcawl commented on specification/security/suggest_user_profiles/examples/request/SecuritySuggestUserProfilesExample1.yaml:
```diff
@@ -0,0 +1,15 @@
+        "u_79HkWkwmnBH5gqFKwoxggWPjEBOur1zLPXQPEl1VBW0_0"
+      ],
+      "labels": {
+        "direction": ["north", "east"]  <3>
```
```suggestion
        "direction": ["north", "east"]
```

lcawl commented on specification/shutdown/put_node/examples/request/ShutdownPutNodeExample1.yaml:
```diff
@@ -0,0 +1,8 @@
+description: An example body for a `PUT /_nodes/USpTGYaBSIKbgSUJR2Z9lg/shutdown` request.
+value: |-
+  {
+    "type": "restart", <1>
```
```suggestion
    "type": "restart",
```

lcawl commented on specification/simulate/ingest/examples/request/SimulateIngestExample1.yaml:
```diff
@@ -0,0 +1,66 @@
+        }
+      }
+    ],
+    "pipeline_substitutions": { <1>
```
```suggestion
    "pipeline_substitutions": {
```

lcawl commented on specification/slm/put_lifecycle/examples/request/SlmPutLifecycleExample1.yaml:
```diff
@@ -0,0 +1,17 @@
+description: An example body for a `PUT _slm/policy/nightly-snapshots` request.
+value: |-
+  {
+    "schedule": "0 30 1 * * ?",       <1>
```
```suggestion
    "schedule": "0 30 1 * * ?",
```

lcawl commented on specification/slm/put_lifecycle/examples/request/SlmPutLifecycleExample1.yaml:
```diff
@@ -0,0 +1,17 @@
+value: |-
+  {
+    "schedule": "0 30 1 * * ?",       <1>
+    "name": "<nightly-snap-{now/d}>", <2>
```
```suggestion
    "name": "<nightly-snap-{now/d}>",
```

lcawl commented on specification/slm/put_lifecycle/examples/request/SlmPutLifecycleExample1.yaml:
```diff
@@ -0,0 +1,17 @@
+  {
+    "schedule": "0 30 1 * * ?",       <1>
+    "name": "<nightly-snap-{now/d}>", <2>
+    "repository": "my_repository",    <3>
```
```suggestion
    "repository": "my_repository",
```

lcawl commented on specification/slm/put_lifecycle/examples/request/SlmPutLifecycleExample1.yaml:
```diff
@@ -0,0 +1,17 @@
+    "name": "<nightly-snap-{now/d}>", <2>
+    "repository": "my_repository",    <3>
+    "config": {
+      "indices": "*",                 <4>
```
```suggestion
      "indices": "*",
```

lcawl commented on specification/slm/put_lifecycle/examples/request/SlmPutLifecycleExample1.yaml:
```diff
@@ -0,0 +1,17 @@
+    "repository": "my_repository",    <3>
+    "config": {
+      "indices": "*",                 <4>
+      "include_global_state": true    <5>
```
```suggestion
      "include_global_state": true
```

lcawl commented on specification/slm/put_lifecycle/examples/request/SlmPutLifecycleExample1.yaml:
```diff
@@ -0,0 +1,17 @@
+      "indices": "*",                 <4>
+      "include_global_state": true    <5>
+    },
+    "retention": {                    <6>
```
```suggestion
    "retention": {
```

lcawl commented on specification/watcher/put_watch/examples/request/WatcherPutWatchExample1.yaml:
```diff
@@ -0,0 +1,20 @@
+value: |-
+  {
+    "trigger" : {
+      "schedule" : { "interval" : "10s" } <1>
```
```suggestion
      "schedule" : { "interval" : "10s" }
```

miguelgrinberg commented:
The `method_request` is only used in request (after this PR is merged). In the previous PR I added it to requests that only had responses (due to not have a request body), but now that this limitation was lifted it makes more sense to always store these in the request examples. I have moved the `method_request` field out of all these responses and created new examples w/o body for them. This is also included in this PR.

So to summarize. The `method_request` field should be used only in requests. It is ignored in responses. In a request example, it is valid to have a `method_request` field and omit the `value` field (the body).

And thanks for catching the callouts! I will do another pass to filter those out.
